config/config.go

```diff
@@ -165,16 +168,39 @@ func GetConfig(configPath *string) (*Config, error) {
 		Database: rawCfg.Database,
 	}, nil
 }
-
 func readConfigFile(configPath *string) (*rawConfig, error) {
 	var rawCfg rawConfig
-	file, err := ioutil.ReadFile(*configPath)
+	// read main config file.
+	mainFileByte, err := ioutil.ReadFile(*configPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "read config file")
+	}
+	if _, err := toml.Decode(string(mainFileByte), &rawCfg); err != nil {
+		return nil, errors.Wrap(err, "read config file")
+	}
+	// if no sub config file
+	if rawCfg.Include == "" {
+		return &rawCfg, nil
+	}
+	// read sub config files.
+	dir, err := ioutil.ReadDir(rawCfg.Include)
 	if err != nil {
-		// err
 		return nil, errors.Wrap(err, "read config file")
 	}
-	// json.Unmarshal(file, &rawCfg)
-	if _, err := toml.Decode(string(file), &rawCfg); err != nil {
+	// merge config.
+	buffer := [][]byte{mainFileByte}
+	for _, f := range dir {
+		if !f.IsDir() {
+			realPath := path.Join(rawCfg.Include, f.Name())
+			fileByte, err := ioutil.ReadFile(realPath)
+			if err != nil {
+				return nil, errors.Wrap(err, "read config file")
+			}
+			buffer = append(buffer, fileByte)
+		}
+	}
+	confBody := bytes.Join(buffer, []byte{})
```

Comment:
When I remove the newline in the end of `config/repositories/repo_1.toml`, it'll got `branch = ["master", "release*"][[repo]]` after join the contents, the test will fail. This will make toml parse failed. You may manually add newline for each file to avoid this accident.

Reply:
updated.